File: fragment/BlockTypeFragment.ts
import { HTMLText, Text } from "pixi.js";
import Fragment, { decode, FragmentType, register } from "./Fragment";

//@ts-ignore
import * as wasm from "../WasmEndec-1.0-SNAPSHOT/js/endec";



const BLOCK_TYPE = register("trickster:block_type", 0x44aa33, (object: any) => {
    if (object instanceof wasm.BlockTypeFragment) {
        return new BlockTypeFragment(object.block);
    }
    return null;
});

export default class BlockTypeFragment extends Fragment {
    blockType: string;

    constructor(blockType: string) {
        super();

        this.blockType = blockType;
    }

    override toString(): string {
        return this.blockType
    }

    override type(): FragmentType {
        return BLOCK_TYPE;
    }
}

File: site/src/jsMain/kotlin/me/maplesyrum/tricksterStudio/spell/fragment/BlockTypeFragment.kt
package me.maplesyrum.tricksterStudio.spell.fragment

import me.maplesyrum.tricksterStudio.Identifier
import me.maplesyrum.tricksterStudio.external.pixi.HTMLText
import tree.maple.kendec.impl.StructEndecBuilder


class BlockTypeFragment(
    val blockType: Identifier
) : Fragment() {
    override fun type(): FragmentType<*> = FragmentType.BLOCK_TYPE

    companion object {
        val ENDEC = StructEndecBuilder.of(
            Identifier.ENDEC.fieldOf("block", BlockTypeFragment::blockType),
            ::BlockTypeFragment
        )
    }

    override fun toString(): String = blockType.toString()
}